File: Ohter/UnitTest/UnitTest-my-demo/app/src/main/java/com/saltwater/junitdempo/MainActivity.java
package com.saltwater.junitdempo;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void testFunction(){
        EditText editText = findViewById(R.id.et_test);
        if (editText.length()==11){
            editText.setText("666");
        }
    }


}

File: Base/IntenFilter/app/src/main/java/com/saltwater/intenfilterdemo/MainActivity.kt
package com.saltwater.intenfilterdemo

import android.content.Intent
import android.net.Uri
import android.support.v7.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        var intent = Intent("test_action")
        intent.setDataAndType(Uri.parse("http://cctv"),"video/mpeg")
        intent.resolveActivity()
        startActivity(intent)

    }
}
